NXDOC-2909 Release Notes for WebUI 3-1-27 (#2621)
nuxeo/doc.nuxeo.com-platform-spaces: src/nxdoc/web-ui/web-ui-release-notes/web-ui-release-notes-3-1-27.md

---
title: Version 3.1.27
description: Discover what's new in Web UI 3.1.27.
review:
  comment: ''
  date: '2026-02-11'
  status: ok
toc: true
labels:
tree_item_index: 974
hidden: true
---

{{{multiexcerpt 'matching-notes' page='web-ui-release-notes'}}}

{{! multiexcerpt name='web-ui-updates'}}

## What’s New in Web UI for LTS 2023 (Version 3.1.27)

**Bug Fixes & Enhancements** 

- ***Normal reload for custom messages and templates:***

>   - Internationalization (i18n) resources under /ui/i18n/ now bypass caching, ensuring the latest translations are always displayed on normal reload and doesnt require hard reload.

Reviewer: ```suggestion
  - Internationalization (i18n) resources under /ui/i18n/ now bypass caching, ensuring the latest translations are always displayed on normal reload and do not require hard reload.
```

>   - Updated Service Worker behavior to prevent caching of Polymer HTML imports which ensures newly added pages and custom slots to load correctly after browser reloads.

Reviewer: ```suggestion
  - Updated Service Worker behavior to prevent caching of Polymer HTML imports, which ensures newly added pages and custom slots to load correctly after browser reloads.
```

- ***Configurable Confirmation Dialogs for Standard Actions:***

>   - Delete blob functionality now supports configurable confirmation messages which can be enabled/disabled by admins.

Reviewer: ```suggestion
  - Delete blob functionality now supports configurable confirmation messages, which can be enabled/disabled by administrators.
```

- ***Automatic Production Catalog Generation:***

>   - Releasing new WebUI versions now triggers automated production catalog generation, streamlining release workflows and reducing manual steps.

Reviewer: ```suggestion
  - Releasing new WebUI versions now trigger automated production catalog generation, streamlining release workflows and reducing manual steps.
```

- ***Saved Search Optimization:***
  - Fixed an issue where the JSON payload used during save/rename operations was incorrectly modified as an array of strings.
  - Saved search operations now use the correct payload format and no longer fail silently.

>   - Long saved search names were previously truncated in the Filters dropdown.

Reviewer: ```suggestion
  - Fixed an issue where long saved search names were truncated in the Filters dropdown.
```

- ***Correct Handling of JSON Payload for Saved Searches:***
  - Fixed an issue where the JSON payload used during save/rename operations was incorrectly modified as an array of strings.
  - Saved search operations now use the correct payload format and no longer fail silently.

- ***Saved Search Title Truncation:***
  - Long saved search names were previously truncated in the Filters dropdown.
  - The full saved search title is now displayed, improving readability and usability.

- ***Editing Multi‑Valued Numeric Properties:***
  - Editing multi‑valued Number fields through the WebUI Edit layout previously caused save failures due to incorrect type handling.
  - Values are now correctly treated as numeric types, preventing ArrayStoreException errors and ensuring documents save successfully.

**Security Improvements** 

- ***Client‑Side Redirect/Phishing Mitigation:***
  - Strengthened URL validation for file downloads.
  - Only trusted blob: and absolute http(s) URLs are allowed.
  - Redirects built from untrusted DOM input are now blocked to prevent phishing‑style attacks.

- ***Service Worker Caching Reliability:***

>   - Updated Service Worker behavior to prevent caching of Polymer HTML imports.

Reviewer 2: Should not be part of Security Improvements. Can be put in a separate heading ** Other features/improvements **

  - Ensures newly added USER_MENU pages and custom slots load correctly after browser reloads.

- ***Accessibility Pipeline Fixes:***

>   - Resolved accessibility test pipeline failures by pinning the browser environment to a stable version.

Reviewer 2: This is related to the pipeline. Should not be part of Security Improvements. Can be put in a separate heading ** Other features/improvements **
